### ISSUE TITLE
expected dense_input to have 2 dimensions, but got array with shape (1, 1, 2)

### ISSUE THREAD

**Issue by godpow** [hsahovic/poke-env#435]
I've been trying to create an agent for the last few hours and I can't seem to figure out what's causing the error
`import asyncio
import numpy as np
from gym.spaces import Box, Space
from tensorflow import keras
from rl.agents.dqn import DQNAgent
from rl.memory import SequentialMemory
from rl.policy import EpsGreedyQPolicy, LinearAnnealedPolicy
from poke_env.environment.abstract_battle import AbstractBattle
from poke_env.environment import pokemon_type
from poke_env.environment.pokemon_type import PokemonType

from poke_env.player import (
    Gen8EnvSinglePlayer,
    RandomPlayer,
    wrap_for_old_gym_api,
)
from poke_env.player.openai_api import ObsType
class AdvancedDQNPlayer(Gen8EnvSinglePlayer):
    def __init__(self, *args, **kwargs):
        super(AdvancedDQNPlayer, self).__init__(*args, **kwargs)
        self.memory = []
        self.gamma = 0.9
        self.epsilon = 1.0
        self.epsilon_decay = 0.995
        self.model = self.load_or_build_model()

    def reset(self, *args, **kwargs):
        try:
            return super(AdvancedDQNPlayer, self).reset(*args, **kwargs)
        except TypeError as e:
            print(f"TypeError caught: {e}")
            raise
    def load_or_build_model(self):
        try:
            return keras.models.load_model("advanced_dqn_model.h5")
        except:
            return self.build_model()

    def build_model(self):
        model = keras.models.Sequential([
            keras.layers.Dense(64, activation='relu', input_shape=(2,)),  # Update the input shape here
            keras.layers.Dense(64, activation='relu'),
            keras.layers.Dense(self.action_space.n, activation='linear')
        ])
        model.compile(optimizer=keras.optimizers.Adam(learning_rate=0.001),
                      loss='mse')
        return model

    def calc_reward(self, last_battle, current_battle) -> float:
        return self.reward_computing_helper(
            current_battle, fainted_value=2.0, hp_value=1.0, victory_value=30.0
        )

    def embed_battle(self, battle: AbstractBattle) -> ObsType:
        moves_base_power = -np.ones(4)
        moves_dmg_multiplier = np.ones(4)
        type_chart = {
            "NORMAL": {"ROCK": 0.5, "GHOST": 0, "STEEL": 0.5},
            "FIRE": {"FIRE": 0.5, "WATER": 0.5, "GRASS": 2, "ICE": 2, "BUG": 2, "ROCK": 0.5, "DRAGON": 0.5, "STEEL": 2},
            "WATER": {"FIRE": 2, "WATER": 0.5, "GRASS": 0.5, "GROUND": 2, "ROCK": 2, "DRAGON": 0.5},
            "ELECTRIC": {"WATER": 2, "ELECTRIC": 0.5, "GRASS": 0.5, "GROUND": 0, "FLYING": 2, "DRAGON": 0.5},
            "GRASS": {"FIRE": 0.5, "WATER": 2, "GRASS": 0.5, "POISON": 0.5, "GROUND": 2, "FLYING": 0.5, "BUG": 0.5,
                      "ROCK": 2, "DRAGON": 0.5, "STEEL": 0.5},
            "ICE": {"FIRE": 0.5, "WATER": 0.5, "GRASS": 2, "ICE": 0.5, "GROUND": 2, "FLYING": 2, "DRAGON": 2,
                    "STEEL": 0.5},
            "FIGHTING": {"NORMAL": 2, "ICE": 2, "POISON": 0.5, "FLYING": 0.5, "PSYCHIC": 0.5, "BUG": 0.5, "ROCK": 2,
                         "GHOST": 0, "DARK": 2, "STEEL": 2, "FAIRY": 0.5},
            "POISON": {"GRASS": 2, "POISON": 0.5, "GROUND": 0.5, "ROCK": 0.5, "GHOST": 0.5, "STEEL": 0, "FAIRY": 2},
            "GROUND": {"FIRE": 2, "ELECTRIC": 2, "GRASS": 0.5, "POISON": 2, "FLYING": 0, "BUG": 0.5, "ROCK": 2,
                       "STEEL": 2},
            "FLYING": {"ELECTRIC": 0.5, "GRASS": 2, "FIGHTING": 2, "BUG": 2, "ROCK": 0.5, "STEEL": 0.5},
            "PSYCHIC": {"FIGHTING": 2, "POISON": 2, "PSYCHIC": 0.5, "DARK": 0, "STEEL": 0.5},
            "BUG": {"FIRE": 0.5, "GRASS": 2, "FIGHTING": 0.5, "POISON": 0.5, "FLYING": 0.5, "PSYCHIC": 2, "GHOST": 0.5,
                    "DARK": 2, "STEEL": 0.5, "FAIRY": 0.5},
            "ROCK": {"FIRE": 2, "ICE": 2, "FIGHTING": 0.5, "GROUND": 0.5, "FLYING": 2, "BUG": 2, "STEEL": 0.5},
            "GHOST": {"NORMAL": 0, "PSYCHIC": 2, "GHOST": 2, "DARK": 0.5},
            "DRAGON": {"DRAGON": 2, "STEEL": 0.5, "FAIRY": 0},
            "DARK": {"FIGHTING": 0.5, "PSYCHIC": 2, "GHOST": 2, "DARK": 0.5, "FAIRY": 0.5},
            "STEEL": {"FIRE": 0.5, "WATER": 0.5, "ELECTRIC": 0.5, "ICE": 2, "ROCK": 2, "STEEL": 0.5, "FAIRY": 2},
            "FAIRY": {"FIRE": 0.5, "FIGHTING": 2, "POISON": 0.5, "DRAGON": 2, "DARK": 2, "STEEL": 0.5},
        }
        for i, move in enumerate(battle.active_pokemon.moves.values()):
            type_1 = battle.opponent_active_pokemon.type_1.name
            type_2 = battle.opponent_active_pokemon.type_2.name if battle.opponent_active_pokemon.type_2 else None
            move_type = move.type.name

            # Initialize damage multiplier
            dmg_multiplier = 1.0

            # Check type_1 multiplier
            if move_type in type_chart and type_1 in type_chart[move_type]:
                dmg_multiplier *= type_chart[move_type][type_1]

            # Check type_2 multiplier
            if type_2 and move_type in type_chart and type_2 in type_chart[move_type]:
                dmg_multiplier *= type_chart[move_type][type_2]

            moves_dmg_multiplier[i] = dmg_multiplier
        fainted_mon_team = len([mon for mon in battle.team.values() if mon.fainted]) / 6
        fainted_mon_opponent = len([mon for mon in battle.opponent_team.values() if mon.fainted]) / 6
        final_vector = np.concatenate(
            [moves_base_power, moves_dmg_multiplier, [fainted_mon_team, fainted_mon_opponent]]
        )
        return np.float32(final_vector)

    def describe_embedding(self) -> Space:
        low = [-1, -1, -1, -1, 0, 0, 0, 0, 0, 0]
        high = [3, 3, 3, 3, 4, 4, 4, 4, 1, 1]
        return Box(
            np.array(low, dtype=np.float32),
            np.array(high, dtype=np.float32),
            dtype=np.float32,
        )

async def main():
    opponent = RandomPlayer(battle_format="gen8randombattle")
    env = AdvancedDQNPlayer(
        battle_format="gen8randombattle",
        opponent=opponent,
        start_challenging=True
    )
    env = wrap_for_old_gym_api(env)

    n_action = env.action_space.n
    input_shape = (1,) + env.observation_space.shape  # This line remains unchanged
    model = env.model

    memory = SequentialMemory(limit=10000, window_length=1)
    policy = LinearAnnealedPolicy(
        EpsGreedyQPolicy(),
        attr="eps",
        value_max=1.0,
        value_min=0.05,
        value_test=0.0,
        nb_steps=10000,
    )

    dqn = DQNAgent(
        model=model,
        nb_actions=n_action,
        policy=policy,
        memory=memory,
        nb_steps_warmup=1000,
        gamma=0.5,
        target_model_update=1,
        delta_clip=0.01,
        enable_double_dqn=True,
    )
    dqn.compile(keras.optimizers.Adam(learning_rate=0.00025), metrics=["mae"])

    dqn.fit(env, nb_steps=10000)
    env.close()

if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(main())
`
